data-prepper-plugins/cloudwatch-metrics-source/src/test/java/org/opensearch/dataprepper/plugins/source/CloudwatchMetricsSourceTest.java

```diff
@@ -0,0 +1,86 @@
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+public class CloudwatchMetricsSourceTest {
+
+    private final String PLUGIN_NAME = "cloudwatch";
+    private final String TEST_PIPELINE_NAME = "cloudwatch-test-pipeline";
+
+    private CloudwatchMetricsSource cloudwatchMetricsSource;
+    private PluginMetrics pluginMetrics;
+    private CloudwatchMetricsSourceConfig cloudwatchMetricsSourceConfig;
+    private PluginFactory pluginFactory;
+    private AcknowledgementSetManager acknowledgementSetManager;
+    private AwsCredentialsSupplier awsCredentialsSupplier;
+
+    private int recordsToAccumulate = 100;
+
+    private Duration bufferTimeout = Duration.ofSeconds(10);
+
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+
+
+    @BeforeEach
+    void setUp() {
+        pluginMetrics = PluginMetrics.fromNames(PLUGIN_NAME, TEST_PIPELINE_NAME);
+        cloudwatchMetricsSourceConfig = mock(CloudwatchMetricsSourceConfig.class);
+        pluginFactory = mock(PluginFactory.class);
+        acknowledgementSetManager = mock(AcknowledgementSetManager.class);
+        awsCredentialsSupplier = mock(AwsCredentialsSupplier.class);
+        awsAuthenticationOptions = mock(AwsAuthenticationOptions.class);
+        when(awsAuthenticationOptions.getAwsRegion()).thenReturn(Region.AF_SOUTH_1);
+        when(awsAuthenticationOptions.getAwsStsRoleArn()).thenReturn("test-arn");
+        when(cloudwatchMetricsSourceConfig.getAwsAuthenticationOptions()).thenReturn(awsAuthenticationOptions);
+        cloudwatchMetricsSource = new CloudwatchMetricsSource(pluginMetrics, cloudwatchMetricsSourceConfig,awsCredentialsSupplier);
+    }
+
+    @Test
+    void start_should_throw_IllegalStateException_when_buffer_is_null() {
+        assertThrows(IllegalStateException.class, () -> cloudwatchMetricsSource.start(null));
+    }
+
+    private BlockingBuffer<Record<Event>> getBuffer() {
+        final HashMap<String, Object> integerHashMap = new HashMap<>();
+        integerHashMap.put("buffer_size", 2);
+        integerHashMap.put("batch_size", 2);
+        final PluginSetting pluginSetting = new PluginSetting("blocking_buffer", integerHashMap);
+        pluginSetting.setPipelineName("pipeline");
+        return new BlockingBuffer<>(pluginSetting);
+    }
+
+    @Test
+    void start_should_read_data_from_cloud_watch_and_push_to_buffer(){
+        final BlockingBuffer<Record<Event>> buffer = getBuffer();
+        final BufferAccumulator bufferAccumulator = mock(BufferAccumulator.class);
+        try (final MockedStatic<BufferAccumulator> bufferAccumulatorMockedStatic = mockStatic(BufferAccumulator.class)) {
+            bufferAccumulatorMockedStatic.when(() -> BufferAccumulator.create(buffer, recordsToAccumulate, bufferTimeout))
+                    .thenReturn(bufferAccumulator);
+            cloudwatchMetricsSource.start(buffer);
+        }
+        assertThat(buffer.isEmpty(),equalTo(true));
```

Comment:
If the test is asserting empty buffer, that means nothing is being pushed to the buffer, right? Please send N number of records to the source and assert that buffer has N records.